pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -304,32 +304,23 @@ func (s *DiscoveryServer) StreamAggregatedResources(stream discovery.AggregatedD
 				continue
 			}
 
+			if err := s.handleTypeURL(discReq.TypeUrl, con.node); err != nil {
+				return err
+			}
 			switch discReq.TypeUrl {
 			case ClusterType, v3.ClusterType:
-				if err := s.handleTypeURL(discReq.TypeUrl, &con.node.RequestedTypes.CDS); err != nil {
-					return err
-				}
 				if err := s.handleCds(con, discReq); err != nil {
```

Comment:
Yep, I am planning to move as much as I can over - hopefully to the point we can use Generator but not sure yet. If we move the ACK part over it will break things for sure for now so trying to do it piece by piece